lib/submissions.spec.ts

```diff
@@ -168,3 +170,46 @@ describe('patchSubmissionForStep', () => {
     });
   });
 });
+
+describe('approveSubmission', () => {
+  it('should work properly', async () => {
+    mockedAxios.patch.mockResolvedValue({
+      data: { submissionId: '123' },
+    });
+
+    await approveSubmission('foo', 'bar');
+    expect(mockedAxios.patch).toHaveBeenCalled();
+    expect(mockedAxios.patch.mock.calls[0][0]).toEqual(
+      `${ENDPOINT_API}/submissions/foo`
+    );
+    expect(mockedAxios.patch.mock.calls[0][1]).toEqual({
+      editedBy: 'bar',
+      submissionState: 'approved',
+    });
+    expect(mockedAxios.patch.mock.calls[0][2]?.headers).toEqual({
+      'x-api-key': AWS_KEY,
+    });
+  });
+});
+
+describe('returnForEdits', () => {
+  it('should work properly', async () => {
+    mockedAxios.patch.mockResolvedValue({
+      data: { submissionId: '123' },
+    });
+
+    await returnForEdits('foo', 'bar', 'test reason');
+    expect(mockedAxios.patch).toHaveBeenCalled();
+    expect(mockedAxios.patch.mock.calls[0][0]).toEqual(
+      `${ENDPOINT_API}/submissions/foo`
+    );
+    expect(mockedAxios.patch.mock.calls[0][1]).toEqual({
+      editedBy: 'bar',
+      submissionState: 'in_progress',
+      rejectionReason: 'test reason',
+    });
+    expect(mockedAxios.patch.mock.calls[0][2]?.headers).toEqual({
+      'x-api-key': AWS_KEY,
+    });
```

Comment:
This could be a lot tidier using `.toHaveBeenCalledWith()`

Reply:
can you give an example? this pattern is throughout the codebase and i agree it looks a bit ugly. toBeCalledWith doesn't seem to be able to let me choose just the headers object

Reply:
never mind, got it